back/engines/commercial/content_builder/spec/policies/content_builder/layout_policy_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe ContentBuilder::LayoutPolicy do
```

Comment:
I think it could be useful write specs by role (admin, project moderator, folder moderator). Because if someone would change the policy such that project moderators can no longer upsert a layout, I don't think the specs would catch this case.

Reply:
And I think it would also be good to verify that a project moderator of a different project cannot upsert/destroy a layout. We've might have had a few bugs / security issues like that in the past.

Reply:
Let's talk about this, because I disagree:
1. Today the `LayoutPolicy` is only used for `Projects`, but in the future (next iterations) it will not. So tying the specs to a project moderation role will make them invalid later on.
2. The `LayoutPolicy` makes abstraction of what content buildable object is used. It only requires that `UserRoleService#can_moderate?` returns `true`. That method supports more models than `Project` only.
3. Covering all roles could turn into a test maintenance problem. Every time someone would add a role and adapt the `UserRoleService`, all the policies using those roles would have to be adapted too. Given point 2, that is not required.

> if someone would change the policy such that project moderators can no longer upsert a layout, I don't think the specs would catch this case.

When someone changes `LayoutPolicy`, the specs have to be adapted too. The specs describe that `can_moderate?` is allowed to be called, but that is not strong enough. It should be an expectation. I will change the code. So when someone changes the policy, it should be clear from the specs that there is a contract between `LayoutPolicy` and `UserRoleService`, which should not be broken without thinking about the consequences.